### PR TITLE
koordlet: filter out Pod not run by runc to support multiple runtime

### PR DESCRIPTION
**Pull request by jasonliu747**
Signed-off-by: Jason Liu <jasonliu747@gmail.com>

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/koordlet/util/runtimeclass.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// NOTE: functions in this file can be overwritten for extension
+
+// GetRuntimeClassName returns RuntimeClassName which refers to a RuntimeClass object in
+// the node.k8s.io group, which should be used to run this pod.
+func GetRuntimeClassName(pod *corev1.Pod) *string {
+	return pod.Spec.RuntimeClassName
+}
+
+// ShouldIgnorePod will return true if Pod uses RuntimeClass other than runc.
+func ShouldIgnorePod(pod *corev1.Pod) bool {
```
Comment (zwzhang0107) [inline pkg/koordlet/util/runtimeclass.go:32]: since this only works for `metric advisor`, how about mv inside as private(e.g. ignorePodCollection)

Comment (jasonliu747) [reply pkg/koordlet/util/runtimeclass.go]: done

File: pkg/koordlet/metricsadvisor/utils.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+// getRuntimeClassName returns RuntimeClassName which refers to a RuntimeClass object in
+// the node.k8s.io group, which should be used to run this pod.
+func getRuntimeClassName(pod *corev1.Pod) *string {
+	return pod.Spec.RuntimeClassName
+}
+
+// shouldIgnorePod will return true if Pod uses RuntimeClass other than runc.
+func shouldIgnorePod(pod *corev1.Pod) bool {
+	runtimeClass := getRuntimeClassName(pod)
```
Comment (zwzhang0107) [inline pkg/koordlet/metricsadvisor/utils.go:17]: return true directly?

File: pkg/koordlet/metricsadvisor/collector.go [hunk 3]
```diff
@@ -227,6 +227,10 @@ func (c *collector) collectPodResUsed() {
 	podMetas := c.statesInformer.GetAllPods()
 	for _, meta := range podMetas {
 		pod := meta.Pod
+		// koordlet only supports standard OCI-runtime, such as runc.
+		if shouldIgnorePod(pod) {
```
Comment (saintube) [inline pkg/koordlet/metricsadvisor/collector.go:231]: We may check the pod meta (e.g. pod runtimeclass, pod annotations) for the corresponding collector plugin, but not ignore all non-runc pods.

e.g. By default, we keep the collection for runc pods. For a kata-runtime pod, we collect its pod metric via a collector plugin of kata-runtime.

Comment (jasonliu747) [reply pkg/koordlet/metricsadvisor/collector.go:231]: I got your point. 
Although we ignore all non-runc pods here, users are still free to create their own plugins to collect Pods run by other runtimes.

**Comment by zwzhang0107** [comment]
we also need a TODO for "making each collector as an independent plugin in metricsAdvisor for better extension", just like those in statesInformers.

File: pkg/koordlet/metricsadvisor/utils.go (hunk 2 above)
Comment (jasonliu747) [reply pkg/koordlet/metricsadvisor/utils.go:33]: Even though this function might be overwritten, I think we should still keep this simple check. WDYT?

**Comment by jasonliu747** [comment]
> we also need a TODO for "making each collector as an independent plugin in metricsAdvisor for better extension", just like those in statesInformers.

Totally agree with you!

**Pull request closed without merging**